src/opencl/mod.rs

```diff
@@ -167,20 +182,52 @@ pub enum GPUSelector {
 }
 
 impl GPUSelector {
-    pub fn get_bus_id(&self) -> GPUResult<u32> {
+    pub fn get_bus_id(&self) -> Option<u32> {
+        match self {
+            GPUSelector::BusId(bus_id) => Some(*bus_id),
+            GPUSelector::Index(index) => get_device_bus_id_by_index(*index),
+        }
+    }
+
+    pub fn get_device(&self) -> Option<Device> {
         match self {
-            GPUSelector::BusId(bus_id) => Ok(*bus_id),
-            GPUSelector::Index(index) => Ok(get_device_bus_id_by_index(*index)?),
+            GPUSelector::BusId(_bus_id) => todo!(),
```

Comment:
do you plan on leaving this `todo`?

Reply:
I did not plan on it, thanks for the reminder.